### PR TITLE
Adding better error msgs to configuration checks

### PR DESCRIPTION
**Pull request by dlemures**
Adding better error msgs to configuration checks
Also moving from Stack to Set, the stack was not needed and set is more efficient

### CODE REVIEW

File: toothpick-runtime/src/main/java/toothpick/InjectionCycleFormatter.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package toothpick;
+
+import java.util.List;
+
+/**
+ * Formats an injection cycle to its String representation.
+ */
+public class InjectionCycleFormatter {
```
Comment (stephanenicolas) [inline toothpick-runtime/src/main/java/toothpick/InjectionCycleFormatter.java]: make it non static, and I would love to see a few tests to see what the messages look like.

File: toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+public class CyclicDependencyExceptionTest {
+
+  @Test
+  public void newCyclicDependencyException_showGenerateWholePath_whenCycleStartsPath() throws Exception {
+    String expectedErrorMessage =
+        "Class toothpick.data.CyclicFoo creates a cycle:\n"
+            + "\n"
+            + "               ================\n"
+            + "              ||              ||\n"
+            + "              \\/              ||\n"
```
Comment (stephanenicolas) [inline toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java]: there is an extra `\`

File: toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    List<Class> path = new ArrayList<>();
+    path.add(CyclicFoo.class);
+
+    //WHEN
+    new CyclicDependencyException(path, null);
+
+    //THEN
+    fail("Should throw an exception as a wrong parameters are passed");
+  }
+}
```
Comment (stephanenicolas) [inline toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java]: add a line

File: toothpick-runtime/src/main/java/toothpick/CyclicDependencyException.java [hunk 4]
```diff
@@ -19,4 +24,68 @@ public CyclicDependencyException(Throwable cause) {
...
+    for (Class clazz : path) {
+      length = clazz.getName().length();
+      if (length > max) {
+        max = length;
+      }
+    }
+    return max;
+  }
+
+  private static String generateRepeatedCharacterString(char c, int n) {
```
Comment (stephanenicolas) [inline toothpick-runtime/src/main/java/toothpick/CyclicDependencyException.java]: jut `repeat(..)` ?

File: toothpick-runtime/src/main/java/toothpick/CyclicDependencyException.java [hunk 5]
```diff
@@ -19,4 +24,68 @@ public CyclicDependencyException(Throwable cause) {
...
+  private static void addLine(StringBuilder builder, String content, int middleWordPos, int loopLinePosition) {
+    int leftMarginSize = middleWordPos - content.length() / 2;
+    int rightMarginSize = loopLinePosition - leftMarginSize - content.length();
+    builder.append(generateRepeatedCharacterString(' ', leftMarginSize));
+    builder.append(content);
+    builder.append(generateRepeatedCharacterString(' ', rightMarginSize));
+    builder.append("||\n");
+  }
+
+  private static int findLengthLongestClassName(List<Class> path) {
```
Comment (stephanenicolas) [inline toothpick-runtime/src/main/java/toothpick/CyclicDependencyException.java]: `findLengthLongestClassName` -> `findLongestClassNameLength`

**Comment by stephanenicolas** [comment]
RTM cool

File: toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java (hunk 3 above)
Comment (dlemures) [inline toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java]: done

File: toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java (hunk 2 above)
Comment (dlemures) [inline toothpick-runtime/src/test/java/toothpick/CyclicDependencyExceptionTest.java]: Only `\/` is generated. The extra `\` is just for the String representation

**Pull request merged**